### PR TITLE
[PATCH v3] linux-gen: macros: prefix implementation internal macros

### PR DESCRIPTION
**Pull request by MatiasElo**
Add _ODP prefixes to implementation internal helper macros.

Signed-off-by: Matias Elo <matias.elo@nokia.com>

### CODE REVIEW

**Comment by psavol** [comment]
macros_internal.h and align_internal.h could be combined into a single header

**Comment by MatiasElo** [comment]
V2:
- Merged `odp_align_internal.h` contents `odp_macros_internal.h`
- Use `_ODP_ARRAY_SIZE()` macro

File: platform/linux-generic/include/odp_macros_internal.h [hunk 1]
```diff
@@ -17,37 +18,67 @@
...
 	})
 
-#define MAX3(a, b, c) (MAX(MAX((a), (b)), (c)))
+#define _ODP_MAX3(a, b, c) (_ODP_MAX(_ODP_MAX((a), (b)), (c)))
 
-#define odp_container_of(pointer, type, member) \
-	((type *)(void *)(((char *)pointer) - offsetof(type, member)))
+/* Macros to calculate ODP_ROUNDUP_POWER2_U32() in five rounds of shift
+ * and OR operations. */
+#define __ODP_RSHIFT_U32(x, y) (((uint32_t)(x)) >> (y))
```
[the remark this answers is not in the document]
Comment (MatiasElo) [reply platform/linux-generic/include/odp_macros_internal.h:45]: `odp_macros_internal.h` is implementation internal, so unfortunately it cannot be used with example applications.

File: platform/linux-generic/include/odp_macros_internal.h [hunk 2]
```diff
@@ -17,37 +18,67 @@
...
+
+#define _ODP_MIN(a, b)				\
 	__extension__ ({			\
 		__typeof__(a) tmp_a = (a);	\
 		__typeof__(b) tmp_b = (b);	\
 		tmp_a < tmp_b ? tmp_a : tmp_b;	\
 	})
 
-#define MAX(a, b)				\
+#define _ODP_MAX(a, b)				\
```
[the remark this answers is not in the document]
Comment (MatiasElo) [reply platform/linux-generic/include/odp_macros_internal.h:34]: `odp_macros_internal.h` is implementation internal, so unfortunately it cannot be used with example applications.

[the remark this answers is not in the document]
Comment (TuomasTaipale) [reply platform/linux-generic/include/odp_macros_internal.h:34]: Ah, yes, of course.

**Review by TuomasTaipale — APPROVED** [review]

**Pull request merged**